### PR TITLE
Add event.outcome to transactions and spans

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

This PR adds `outcome` as a field of `transaction` and `span`. For RUM v3, `outcome` is abbreviated to `o`.

The value will be recorded as the [`event.outcome`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) in the event documents, and will be used for calculating error rates.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

I have considered changes for:
- [x] documentation
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
- [x] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
~- [ ] telemetry~
- [x] Elasticsearch Service (https://cloud.elastic.co)
- [x] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)
- [x] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)

## How to test these changes

Cannot be manually tested until we have a modified agent.

## Related issues

https://github.com/elastic/apm/pull/299

### CODE REVIEW

File: model/transaction.go [hunk 1]
```diff
@@ -134,6 +135,9 @@ func (e *Transaction) Transform(_ context.Context, _ *transform.Config) []beat.E
 			utility.Set(fields, "url", e.Page.URL.Fields())
 		}
 	}
+	if e.Outcome != "" {
```
Comment (jalvz) [inline model/transaction.go:138]: it isn't the case that at `Outcome` is not empty?

File: tests/system/drop_unsampled_transactions.approved.json [hunk 2]
```diff
@@ -124,9 +126,9 @@
             }
         },
         "observer": {
-            "ephemeral_id": "ebd89158-9f85-469e-9bc0-9e19b174a1ba",
+            "ephemeral_id": "d0a481f0-3688-40dc-8a59-04abdf72261a",
             "hostname": "goat",
-            "id": "01019c57-ff25-4ddf-b9e9-2f421577a9a2",
+            "id": "fb7e85d9-3666-4b9a-8440-1448c17b5738",
```
Comment (jalvz) [inline tests/system/drop_unsampled_transactions.approved.json:131]: why all these changes? `id, `ingested`, etc?

Comment (axw) [reply tests/system/drop_unsampled_transactions.approved.json:131]: They get automatically updated by the approvals tests.

File: model/transaction.go (hunk 1 above)
Comment (axw) [reply model/transaction.go:138]: Yes, good catch. `Outcome` should always be non-empty now.

**Comment by axw** [comment]
Note to reviewer: you might want to filter out the `.json` files when reviewing initially. This change has caused a lot of `.approved.json` files to be updated on account of the server setting a default value for outcome if missing.

**Review by simitt — APPROVED** [review]

**Comment by beniwohli** [comment]
An agent implementation can be found here: https://github.com/elastic/apm-agent-python/pull/899

**Comment by axw** [comment]
Thanks @beniwohli. I tested with apm-agent-python & aiohttp, and it appears to be working well:

image

**Pull request merged**